subprojects/core/src/main/java/org/gradle/plugin/management/internal/InternalPluginResolutionStrategy.java

```diff
@@ -0,0 +1,9 @@
+package org.gradle.plugin.management.internal;
+
+import org.gradle.plugin.management.PluginResolutionStrategy;
+import org.gradle.plugin.use.internal.InternalPluginRequest;
+
+public interface InternalPluginResolutionStrategy extends PluginResolutionStrategy {
+
+    void resolvePluginRequest(InternalPluginRequest pluginRequest);
```

Comment:
This method needs to return a (potentially new) PluginRequest, not mutate the existing one.